### PR TITLE
config: add periodic job that runs upgrades using multistage tests

### PR DESCRIPTION
**Pull request by AlexNPavel**
Add new periodic that runs upgrade test from 4.4 stable to 4.5 nightly using multistage tests.

/cc @stevekuznetsov @bbguimaraes

### CODE REVIEW

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+releases:
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml:1]: Why did you need this upgrade variant config?

Comment (AlexNPavel) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml:1]: Not sure if the difference between `nightly` and `ci` really matters much for the periodic upgrade tests though.

Comment (stevekuznetsov) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml:1]: We do want them to be identical

Comment (stevekuznetsov) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml:1]: I guess we will want a set of `ocp-4.5-ci` and `ocp-4.5-nightly` and `ocp-4.5-prerelease` variants (defined by what it means to be `latest` in each config)

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      product: ocp
+      stream: ci
+      version: "4.5"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: e2e-upgrade-informer
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml:17]: Why isn't there an `OPENSHIFT_UPGRADE=true` env here?

File: ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh [hunk 3]
```diff
@@ -13,7 +13,11 @@ if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
   exit 1
 fi
 
-echo "Installing from release ${RELEASE_IMAGE_LATEST}"
+if [[ "${OPENSHIFT_UPGRADE:-false}" == "false" ]]; then
```
Comment (stevekuznetsov) [inline ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh:16]: Why do we need this here? I'd rather just remove it and have us print this during the install step, no need to have `conf` care if it does nothing with this value

Comment (AlexNPavel) [reply ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh]: Moved the echo to the install step.

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml (hunk 2 above)
Comment (AlexNPavel) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml]: Now it's set.

File: ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml (hunk 1 above)
Comment (AlexNPavel) [reply ci-operator/config/openshift/release/openshift-release-master__ocp-4.5-upgrades.yaml]: Renamed the new variant to `ocp-4.5-ci`

File: ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh [hunk 4]
```diff
@@ -13,8 +13,6 @@ if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
   exit 1
 fi
 
-echo "Installing from release ${RELEASE_IMAGE_LATEST}"
```
Comment (wking) [inline ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh:16]: This seems orthogonal to the thrust of this PR and [the `echo`][1] dates back to `ipi-conf-commands.sh` from #6708.  Can we spin this hunk out into a separate PR and get it landed while this one gets iterated on?

[1]: https://github.com/openshift/release/pull/6708/files#diff-50b0e5c1da4c6746d68864286c7c2876R40

File: ci-operator/step-registry/ipi/conf/ipi-conf-ref.yaml [hunk 5]
```diff
@@ -1,6 +1,9 @@
 ref:
   as: ipi-conf
-  from: base
+  from_image:
+    namespace: origin
+    name: centos
+    tag: '8'
```
Comment (wking) [inline ci-operator/step-registry/ipi/conf/ipi-conf-ref.yaml:6]: Why this change?  `base` seems less opinionated for steps that just need generic POSIX stuff, and it's one less thing to forget to bump.

File: ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh (hunk 4 above)
Comment (wking) [reply ci-operator/step-registry/ipi/conf/ipi-conf-commands.sh:16]: Ah, reading down I see you adding a similar `echo` in `ipi-install-install-commands.sh`, so I'm meh on whether we pull this out into a separate PR or not.  And while I'm talking about `ipi-install-install-commands.sh`, seems like a chance to remind folks that the stutter-removing PR is alive again: #7931.

File: ci-operator/step-registry/ipi/install/install/ipi-install-install-ref.yaml [hunk 6]
```diff
@@ -6,5 +6,8 @@ ref:
     requests:
       cpu: 1000m
       memory: 2Gi
+  env:
+  - name: OPENSHIFT_UPGRADE
+    default: "false"
```
Comment (wking) [inline ci-operator/step-registry/ipi/install/install/ipi-install-install-ref.yaml:11]: Please add [`documentation`][1] :).  Something like:

> Select between ${RELEASE_IMAGE_INITIAL} (true) or ${RELEASE_IMAGE_LATEST} (false) for the installed version.

Not clear to me if `documentation` supports Markdown.  If so, backtick the environment variables?

[2]: https://github.com/openshift/ci-tools/pull/854/files#diff-95aa6a9a4f1daf516567fb2eb073cb3fR1225-R1226

File: ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh [hunk 7]
```diff
@@ -62,12 +62,23 @@ if [[ "${CLUSTER_TYPE}" == gcp ]]; then
...
 fi
 
-test_suite=openshift/conformance/parallel
-if [[ -e "${SHARED_DIR}/test-suite.txt" ]]; then
-    test_suite=$(<"${SHARED_DIR}/test-suite.txt")
-fi
-
-openshift-tests run "${test_suite}" \
-    --provider "${TEST_PROVIDER}" \
+if [[ "${OPENSHIFT_UPGRADE:-false}" == "false" ]]; then
```
Comment (wking) [inline ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh:65]: Hrm.  Not clear to me how much we want to pack into `OPENSHIFT_UPGRADE` here with an opinionated conditional inside the step.  How about #9676 and then setting:

```sh
OPENSHIFT_UPGRADE=true  # for the install switch.
TEST_COMMAND=run-upgrade
TEST_SUITE=all
```

Or maybe even RELEASE_IMAGE=RELEASE_IMAGE_INITIAL and then use indirect expansion:

```bash
$ RELEASE_IMAGE_INITIAL=A
$ RELEASE_IMAGE_LATEST=B
$ RELEASE_IMAGE=RELEASE_IMAGE_INITIAL
$ echo "${!RELEASE_IMAGE}"
A
```

And then setting those variables appropriately when we define the job?

**Comment by ingvagabund** [comment]
@AlexNPavel is this PR still active?

**Comment by hexfusion** [comment]
etcd-team is very interested in this

**Comment by stevekuznetsov** [comment]
superseded by https://github.com/openshift/release/pull/11247

/close
